parsers/cxx/cxx_parser_tokenizer.c

```diff
@@ -1161,7 +1161,45 @@ boolean cxxParserParseNextToken(void)
 				}
 
 				if(szReplacement && *szReplacement)
-					return TRUE; // already have a token to return
+				{
+					// Already have a token to return.
+					// Check again for keywords.
+					
+					// There is a bit of code duplication here but I'd rather
+					// avoid a function call and a switch as the keyword check
+					// must be fast (especially the one above).
```

Comment:
would it be so slow or problematic, really?  at worse, an inlining hint to the compiler should help.
I'm a bit afraid of the code duplication here because it's already fairly non-trivial, so likely to grow even more complex in the future.

If the part handling `__attribute__` differently make it too annoying/complex, maybe just leave this part out, and simply have a `ccxParserLookupKeyword()` that only does the lookup and special case handling.